atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,264 @@
+
+        repo = {
+            'name': 'atomic-reactor-koji-plugin-%s' % self.target,
+            'baseurl': baseurl,
+            'enabled': 1,
+            'gpgcheck': 0,
+        }
+
+        # yum doesn't accept a certificate path in sslcacert - it requires a db with added cert
+        # dnf ignores that option completely
+        # we have to fall back to sslverify=0 everytime we get https repo from brew so we'll surely
+        # be able to pull from it
+
+        if baseurl.startswith("https://"):
+            self.log.info("Ignoring certificates in the repo")
+            repo['sslverify'] = 0
+
+        if self.proxy:
+            self.log.info("Setting yum proxy to %s", self.proxy)
+            repo['proxy'] = self.proxy
+
+        yum_repo = YumRepo(os.path.join(YUM_REPOS_DIR, self.target))
+        path = yum_repo.dst_filename
+        self.log.info("yum repo of koji target: '%s'", path)
+        yum_repo.content = render_yum_repo(repo, escape_dollars=False)
+        self.yum_repos.append(yum_repo)
+
+    def __init__(self, workflow, target=None, repourls=None, inject_proxy=None):
+        """
+        constructor
+
+        :param workflow: DockerBuildWorkflow instance
+        :param target: string, koji target to use as a source
+        :param repourls: list of str, URLs to the repo files
+        :param inject_proxy: set proxy server for this repo
+        """
+        super().__init__(workflow)
+        self.target = target
+
+        self.xmlrpc = None
+        self.pathinfo = None
+        self.proxy = None
+        self.repourls = repourls or []
+        self.inject_proxy = inject_proxy
+        self.yum_repos = []
+        self.allowed_domains = self.workflow.conf.yum_repo_allowed_domains
+        self.include_koji_repo = False
+        self._builder_ca_bundle = None
+        self._ca_bundle_pem = None
+        self._dockerfile = None
+
+    def _inject_into_repo_files(self, build_dir: BuildDir):
+        """Inject repo files into a relative directory inside the build context"""
+        if build_dir.dockerfile.baseimage is None:
+            raise RuntimeError("No FROM line in Dockerfile")
+        host_repos_path = build_dir.path / RELATIVE_REPOS_PATH
+        self.log.info("creating directory for yum repos: %s", host_repos_path)
+        os.mkdir(host_repos_path)
+        allow_repo_dir_in_dockerignore(build_dir.path)
+
+        for repo in self.yum_repos:
+            # Update every repo accordingly in a repofile
+            # input_buf ---- updated ----> updated_buf
+            with StringIO(repo.content.decode()) as input_buf, StringIO() as updated_buf:
+                for line in input_buf:
+                    updated_buf.write(line)
+                    # Apply sslcacert to every repo in a repofile
+                    if line.lstrip().startswith('[') and self._builder_ca_bundle:
+                        updated_buf.write(f'sslcacert=/tmp/{self._ca_bundle_pem}\n')
```

Comment:
this is weird now, because you will have to copy that file to each arch builder ..... 

cc @chmeliik

Reply:
I think `/tmp/{self._ca_bundle_pem}` is the path inside the container. We copy the cert from the build-dir to the container with `ADD {self._ca_bundle_pem} /tmp/{self._ca_bundle_pem}`, so that should still work

Reply:
I double-checked it and yes, it is as @chmeliik said. we copy ca_bundle to /tmp inside container. We are also copying it from its original path to platform-specific build dirs and then providing the relative path inside the build dir so it's definitely there when adding